README.md

```diff
@@ -30,45 +30,78 @@ There is also a Slack channel for discussion and support. Click [here](https://t
 
 ## Development
 
-To install an extension in editable mode, simply change to its subdirectory
-within the `modules` directory, and run:
+This project uses [poetry](https://python-poetry.org/) for packaging and dependency management and
+[Nox](https://nox.thea.codes/en/stable/) for task automation.
+
+### Recommended development setup
+
+Create a python virtual environment within the project root directory:
+
+```shell
+python -m venv .venv
```

Comment:
Do we want to recommend venv? Is there a special need to use venv? Would this not work with the local python installation?

Reply:
It could (and most likely would) work outside a venv, but it is best practice to use venv in order to not break things elsewhere or have dependency problems. Maybe this Nox version requires something that is already installed somewhere else, but the required versions aren't compatible.

Reply:
I think as long as it would work without a venv we should not recommend that. people might want to use something else like anaconda or whatever and might think this is a requirement?

Reply:
Hmmm, I have to disagree 😄 . The information is specifically a _recommendation_, and specifically for `pytket-aqt` developers. If a python developer does't know about python/anaconda environments and what they are for then maybe it's good that they see a recommendation and  have to ask themselves why it's recommended. If they do know about it then they would probably know that it isn't necessarily required and that an anaconda env or no virtual env could also work. In any case, if they aren't in a venv or anaconda env then the section `Local development without Nox` will not work as advertised (because poetry will detect that it isn't in a virtual environment and automatically install everything in an isolated environment, so the commands listed will only work from within that environment).

As far as users of the package goes, they only need to know `pip install pytket-aqt` and nothing about virtual environments.

But if you want I can remove it and fix that one section. Or change the wording so that it's more clear that it isn't necessary.

Reply:
I'll change it and fix the `Local development without Nox section`. Then you are right, it could be less confusing 😄

Reply:
README is updated! (there was also some more outdated stuff there, so cleaned that up as well)